### DISCUSSION TITLE
How to use manually added images from /media/part_images/ as part images?

### DISCUSSION THREAD

**Discussion by WallK** [inventree/InvenTree#4374] [Q&A] [answered]
I have uploaded some images to part_images folder and also created a subfolder there with more images inside
They are in .gif and .webp formats

So how do I attach them to parts now? They are not appearing on the frontend
Maybe there's a way to force folder scan and thumbnail generation?

Thanks!

**Comment by SchrodingersGat** [comment, marked as answer]
@WallK images should be uploaded via the web interface, and not manually added to the `media` directory. There is no way to link images from the media directory - you have to upload them via either the web interface or the API.

You can read more about this here - https://docs.inventree.org/en/latest/part/part/#part-images

As a side note - I am curious why you attempted it this way. Was there something in the documentation which alluded to this being the right approach? :)

**Reply by WallK** [reply]
Firstly, I was reading stable docs, which does not include this part

I have more images than parts and I wanted to sort them into directories
For example, I created PLA folder with all PLA spools images and PETG folder with all PETG images
I don't have (and don't plan to even stock) all the colors and variants, but I have images of all of them scraped from supplier website

I was hoping I could navigate into the folder and pick the right image using the UI image picker
It seemed like a better workflow to me
Especially considering that yellow PLA and yellow PETG images are pretty close but _not the same_ and it will bother me immensely (:

**Reply by SchrodingersGat** [reply]
> Firstly, I was reading stable docs, which does not include this part

I only just added this information :)

In terms of workflow, I'd suggest you look at the the [python interface](https://docs.inventree.org/en/latest/api/python/python/) or [API](https://docs.inventree.org/en/latest/api/api/) to automate the process a bit.

**Reply by WallK** [reply]
I'll look into uploading images programmatically, thanks!
But as far as I understood -- no folders for me, huh?

**Reply by SchrodingersGat** [reply]
No folders for you!

**Reply by WallK** [reply]
I'll go read up on Django, maybe there will be folders for me in the future! (:

**Reply by matmair** [reply]
This workflow is really not how it is intended. You should not touch the server-managed filesystem as there are a lot of security issues that could crop up.
You could built a function to reassign files from a folder - it is not worth the effort for core (unit tests, docs etc. need to be done to keep it stable) in my opinion. I suggest using the API and uploading the files with the python bindings.
